src/WindowsDebugLauncher/DebugLauncher.cs

```diff
@@ -0,0 +1,233 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Security.Principal;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WindowsDebugLauncher
+{
+    internal class DebugLauncher : IDisposable
+    {
+        private static readonly int BUFFER_SIZE = 1024 * 4;
+
+        internal class LaunchParameters
+        {
+            public string PipeServer { get; set; }
+            public string StdInPipeName { get; set; }
+            public string StdOutPipeName { get; set; }
+            public string StdErrPipeName { get; set; }
+            public string PidPipeName { get; set; }
+            public string DbgExe { get; set; }
+            public List<string> DbgExeArgs { get; set; }
+
+            public LaunchParameters()
+            {
+                DbgExeArgs = new List<string>();
+            }
+
+            public bool ValidateParameters()
+            {
+                return !(string.IsNullOrEmpty(PipeServer)
+                    || string.IsNullOrEmpty(StdInPipeName)
+                    || string.IsNullOrEmpty(StdOutPipeName)
+                    || string.IsNullOrEmpty(StdErrPipeName)
+                    || string.IsNullOrEmpty(PidPipeName)
+                    || string.IsNullOrEmpty(DbgExe));
+            }
+
+            public string DbgExeArgsAsString()
+            {
+                StringBuilder argString = new StringBuilder();
+                foreach (var arg in DbgExeArgs.ToList())
+                {
+                    argString.Append(arg);
```

Comment:
>arg [](start = 37, length = 3)

Do you need to do any quoting?

Reply:
Wasn't planning on it? but i can if it contains a space.